tests/test_recipe_yaml_parsing.py

```diff
@@ -162,3 +170,80 @@ def test_parse_munged_run_export_slepc_weak_strong():
     }
+
+
+class RecipeEnvironment:
+    """Manages a temporary environment for recipe testing."""
+
+    def __init__(
+        self, recipe_path: Path, ci_support_files: list[Path] | None = None
+    ) -> None:
+        """
+        Initialize recipe test environment.
+
+        Args:
+            recipe_path: Path to the recipe YAML file
+            ci_support_files: Optional sequence of CI support configuration files
+        """
+        if not recipe_path.is_file():
+            raise FileNotFoundError(f"Recipe file not found: {recipe_path}")
+
+        self.recipe_path = recipe_path
+        self.tmp_dir = TemporaryDirectory()
+        self.root = Path(self.tmp_dir.name)
+
+        self._create_directory_structure()
+        self._setup_recipe()
+        self._setup_ci_support(ci_support_files)
+
+    def _create_directory_structure(self) -> None:
+        """Create the required directory structure."""
+        (self.root / "recipe").mkdir(parents=True, exist_ok=True)
+        (self.root / ".ci_support").mkdir(parents=True, exist_ok=True)
+
+    def _setup_recipe(self) -> None:
+        """Copy recipe file to temporary directory."""
+        try:
+            shutil.copy2(self.recipe_path, self.root / "recipe" / "recipe.yaml")
+        except OSError as e:
+            raise OSError(f"Failed to copy recipe file: {e}")
+
+    def _setup_ci_support(self, ci_support_files: list[Path] | None) -> None:
+        """Setup CI support configuration files."""
+        if ci_support_files:
+            for file in ci_support_files:
+                if not file.is_file():
+                    raise FileNotFoundError(f"CI support file not found: {file}")
+                try:
+                    shutil.copy2(file, self.root / ".ci_support" / file.name)
+                except OSError as e:
+                    raise OSError(f"Failed to copy CI support file {file}: {e}")
+        else:
+            default_config = self.root / ".ci_support" / "linux_64_.yaml"
+            default_config.write_text("target_platform:\n- linux-64\n")
+
+
+@pytest.fixture
+def recipe_env(tmp_path: Path) -> Generator[RecipeEnvironment, None, None]:
+    """Fixture that provides a RecipeEnvironment instance."""
+    recipe_path = Path("tests/test_recipe_yaml/libssh.yaml")
+    env = RecipeEnvironment(recipe_path)
+    yield env
+    env.tmp_dir.cleanup()
+
+
+@pytest.mark.recipe
+class TestRecipeYamlParsing:
+    @pytest.mark.parametrize("recipe_name", ["libssh", "torchvision-reduced"])
+    def test_populate_feedstock_attributes(self, recipe_env, recipe_name):
+        """Test parsing different recipe files."""
+        recipe_yaml = recipe_env.recipe_path.parent / f"{recipe_name}.yaml"
+        subgraph = {}
+        populate_feedstock_attributes(
+            recipe_name,
+            subgraph,
+            recipe_yaml=recipe_yaml.read_text(),
+            feedstock_dir=recipe_env.root,
+        )
+
+        assert subgraph["feedstock_name"] == recipe_name
```

Comment:
A specific test that the recipes comes out without dicts needs to be added here.

Reply:
I added a test to check that all elements of the `set` are str. But that's anyways a given because dicts aren't hashable (the origin of the problem! :)

<img width="234" alt="Screenshot 2025-01-15 at 14 13 17" src="https://github.com/user-attachments/assets/8db0c89e-fd45-45f8-982c-853dbaec5379" />